FreeRTOS/Labs-Project-ota-example-for-aws-iot-core
# Parser callback removal

### Description of Changes
Remove Jobs parser callback

### Manual testing steps
Ran demo. Output below...
```
D Transport_OpenSSL: Successfully imported root CA.
D Transport_OpenSSL: Successfully imported client certificate.
D Transport_OpenSSL: Successfully imported client certificate private key.
D Transport_OpenSSL: Setting server name a77n9d666fj2p-ats.iot.us-east-1.amazonaws.com for SNI.
D Transport_OpenSSL: Established a TLS connection.
Successfully connected to IoT Core
Data topic is $aws/things/DevDesktopThing/streams/AFR_OTA-fe818141-a933-4de9-bc70-deb15234350c/data/json
Get stream topic is $aws/things/DevDesktopThing/streams/AFR_OTA-fe818141-a933-4de9-bc70-deb15234350c/get/json
Handled? 1SUBACK received with packet id: 2
MQTT streams handling incoming message 
Incoming Publish Topic Length: 90 Name: $aws/things/DevDesktopThing/streams/AFR_OTA-fe818141-a933-4de9-bc70-deb15234350c/data/json matches subscribed topic.
Incoming Publish Message length: 43 Message: {"f":0,"i":0,"l":12,"p":"aGVsbG8gd29ybGQK"}
Downloaded Data hello world
 
OTA Completed successfully!
MQTT streams handling incoming message 
Incoming Publish Topic Name: $aws/things/DevDesktopThing/jobs/AFR_OTA-AFR_OTA-MacDevThingOTA2/update/accepted{"timestamp":1691780483}"l":12,"p":"aGVsbG8gd29ybGQK"}A-MacDevThingOTA2","status":"IN_PROGRESS","queuedAt":1691780478,"startedAt":1691780483,"lastUpdatedAt":1691780483,"versionNumber":2,"executionNumber":1,"jobDocument":{"afr_ota":{"protocols":["MQTT"],"streamname":"AFR_OTA-fe818141-a933-4de9-bc70-deb15234350c","files":[{"filepath":"/path","filesize":12,"fileid":0,"certfile":"Code Verify Key","sig-sha256-ecdsa":"MEUCICttjmQMnEqJc5STgGyber0vWjLgLVbopBBx1kInfcOnAiEA2O1+QAqc72lPTDG1W4YETx47PblliL1EtsMphPZ3B0g="}]}}}} does not match subscribed topic.
Unrecognized incoming MQTT message received on topic: $aws/things/DevDesktopThing/jobs/AFR_OTA-AFR_OTA-MacDevThingOTA2/update/accepted
Message: {"timestamp":1691780483}
Unhandled incoming PUBLISH received on topic, message: $aws/things/DevDesktopThing/jobs/AFR_OTA-AFR_OTA-MacDevThingOTA2/update/accepted
{"timestamp":1691780483}
```

### Unit tests
Added. 100% line coverage, 88.9% branch coverage. All missing branch coverage is unreachable.

### Backward compatibility of data and changes
#### Is the change rollback friendly? Think about state in DynamoDB or other resources created by the new code and whether the old code can handle it. 

Manual testing suggests there is no backwards feature regression. I expect this to be backwards compatible.
